lib/include/elements/element/range_slider.hpp

```diff
@@ -0,0 +1,123 @@
+#if !defined(ELEMENTS_RANGE_SLIDER_NOVEMBER_30_2023)
+#define ELEMENTS_RANGE_SLIDER_NOVEMBER_30_2023
+
+// #include <elements/element/slider.hpp>
+#include <elements/element/proxy.hpp>
+#include <elements/element/tracker.hpp>
+#include <elements/view.hpp>
+#include <elements/support.hpp>
+#include <infra/support.hpp>
+
+#include <cmath>
+#include <functional>
+#include <type_traits>
+
+namespace cycfi { namespace elements
+{
+   ////////////////////////////////////////////////////////////////////////////
+   // Sliders
+   ////////////////////////////////////////////////////////////////////////////
+   class range_slider_base : public tracker<>, public receiver<std::pair<double, double>>
+   {
+   using val_type = std::pair<double, double>;
+   public:
+
+                              range_slider_base(std::pair<double, double> init_value)
+                               : _value(init_value)
+                              {}
+
+      view_limits             limits(basic_context const& ctx) const override;
+      void                    draw(context const& ctx) override;
+      void                    layout(context const& ctx) override;
+
+      bool                    scroll(context const&, point, point) override { return false; } //! true or false?
+      void                    begin_tracking(context const& ctx, tracker_info& track_info) override;
+      void                    keep_tracking(context const& ctx, tracker_info& track_info) override;
+      void                    end_tracking(context const& ctx, tracker_info& track_info) override;
+
+      val_type                value() const override;
+      void                    value(val_type val) override;
+      double                  value_first() const { return _value.first; }
+      double                  value_second() const { return _value.second; }
+      void                    value_first(double val) { value({val, _value.second});}
+      void                    value_second(double val) { value({_value.first, val});}
+      void                    edit(view& view_, val_type val) override;
+      virtual void            edit_value_first(double val) { value({val, _value.second});}
+      virtual void            edit_value_second(double val) { value({_value.first, val});}
+      void                    edit_value(val_type val) {edit_value_first(val.first); edit_value_second(val.second);}
+
+      rect                    track_bounds(context const& ctx) const;
+      std::pair<rect, rect>   thumb_bounds(context const& ctx) const;
+
+      virtual std::pair<std::reference_wrapper<element const>, std::reference_wrapper<element const>> thumb() const = 0;
```

Comment:
I'd use using typedefs here to keep the formatting tidy, maybe:

using element_ref_pair = std::pair<std::reference_wrapper<element const>, std::reference_wrapper<element const>>;

element_ref_pair       thumb() const = 0;